puan/__init__.py

```diff
@@ -26,6 +26,41 @@ class Dtype(str, Enum):
 
 class Proposition:
 
+    # @property
```

Comment:
All classes inheriting from Proposition are supposed to have `id` and `bounds`. But due to something it is not possible to declare this way. One thing is to let Proposition inherit `dataclasses.dataclass` but that resulted in other issues. So I thought I'd let it be there out commented but maybe better to remove (since all other classes have id and bounds anyway)